cloudwatch_for_cdm.tf

```diff
@@ -0,0 +1,52 @@
+# ------------------------------------------------------------------------------
+# Provision the IAM policy that allows access the CDM CloudWatch data
+# and attach it to the existing IAM role that the CDM user can already
+# assume to access the CDM CloudTrail data.
+# ------------------------------------------------------------------------------
+
+# Policy document that assigns the necessary permissions to access the
+# CDM CloudWatch data.
+data "aws_iam_policy_document" "allow_access_to_selected_cloudwatch_logs" {
+  statement {
+    actions = [
+      "logs:GetLogEvents",
+    ]
+    effect = "Allow"
+    resources = [
+      # There is no need to adjust the FQDNs for staging, since COOL
+      # staging is not connected to CDM; this is because CDM cannot
+      # handle the fact that COOL production and staging use the same
+      # IPv4 IP space.
+      "arn:aws:logs:${var.aws_region}:${local.sharedservices_account_id}:log-group:/instance-logs/ipa0.cool.cyber.dhs.gov:log-stream:messages",
+      "arn:aws:logs:${var.aws_region}:${local.sharedservices_account_id}:log-group:/instance-logs/ipa0.cool.cyber.dhs.gov:log-stream:secure",
+      "arn:aws:logs:${var.aws_region}:${local.sharedservices_account_id}:log-group:/instance-logs/ipa0.cool.cyber.dhs.gov:log-stream:audit",
+      "arn:aws:logs:${var.aws_region}:${local.sharedservices_account_id}:log-group:/instance-logs/ipa1.cool.cyber.dhs.gov:log-stream:messages",
+      "arn:aws:logs:${var.aws_region}:${local.sharedservices_account_id}:log-group:/instance-logs/ipa1.cool.cyber.dhs.gov:log-stream:secure",
+      "arn:aws:logs:${var.aws_region}:${local.sharedservices_account_id}:log-group:/instance-logs/ipa1.cool.cyber.dhs.gov:log-stream:audit",
+      "arn:aws:logs:${var.aws_region}:${local.sharedservices_account_id}:log-group:/instance-logs/ipa2.cool.cyber.dhs.gov:log-stream:messages",
+      "arn:aws:logs:${var.aws_region}:${local.sharedservices_account_id}:log-group:/instance-logs/ipa2.cool.cyber.dhs.gov:log-stream:secure",
+      "arn:aws:logs:${var.aws_region}:${local.sharedservices_account_id}:log-group:/instance-logs/ipa2.cool.cyber.dhs.gov:log-stream:audit",
+      "arn:aws:logs:${var.aws_region}:${local.sharedservices_account_id}:log-group:/instance-logs/vpn:log-stream:messages",
+      "arn:aws:logs:${var.aws_region}:${local.sharedservices_account_id}:log-group:/instance-logs/vpn:log-stream:secure",
+      "arn:aws:logs:${var.aws_region}:${local.sharedservices_account_id}:log-group:/instance-logs/vpn:log-stream:audit",
```

Comment:
Thoughts on making all of the log groups a `local` and building the list dynamically? Something like:

```terraform
[ for group in local.cloudwatch_log_groups : format("arn:aws:logs:%s:%d:log-group:%s", var.aws_region, local.sharedservices_account_id, group) ]
```

Reply:
I like @mcdonnnj's suggestion - it will make this code a little bit cleaner.  

What I would really like to see is a way for us to remove the hard-coded names of the instances here.  I suppose we could make them Terraform vars or grab them from the outputs of https://github.com/cisagov/cool-sharedservices-freeipa and https://github.com/cisagov/cool-sharedservices-openvpn, but this is complicated by the fact that the log groups for the IPA instances have FQDNs, while those for the OpenVPN instance do not.

Reply:
Please see commit 1f279bd.

Reply:
How about a middle ground in the interest of separating configuration from code- can you make `cdm_instances` into a Terraform variable?  If you want to go a step further, I suppose `cdm_log_streams` could also benefit from being a Terraform variable, but I'll leave that one up to you.

Reply:
Please see commit 691261b.

Reply:
Love it, thanks!